packages/netlify-cms-core/src/__tests__/backend.spec.js

```diff
@@ -0,0 +1,91 @@
+import { resolveBackend } from '../backend';
+import registry from 'Lib/registry';
+
+jest.mock('Lib/registry');
+
+const configWrapper = inputObject => ({
+  get: prop => inputObject[prop],
+});
+
+describe('Backend', () => {
+  describe('filterEntries', () => {
+    let backend;
+
+    beforeEach(() => {
+      registry.getBackend.mockReturnValue({
+        init: jest.fn(),
+      });
+      backend = resolveBackend({
+        getIn: jest.fn().mockReturnValue('git-gateway'),
+      });
+    });
+
+    it('filters string values', () => {
+      const result = backend.filterEntries(
+        {
+          entries: [
+            {
+              data: {
+                testField: 'testValue',
+              },
+            },
+          ],
+        },
+        configWrapper({ field: 'testField', value: 'testValue' }),
+      );
+
+      expect(result.length).toBe(1);
+    });
+
+    it('filters number values', () => {
+      const result = backend.filterEntries(
+        {
+          entries: [
+            {
+              data: {
+                testField: 42,
+              },
+            },
```

Comment:
First of all, **thank you** for adding tests!

I'm thinking we should add an extra entry to each of these `entries` arrays so that it's possible for more than one value to be returned and actual filtering takes place in the tests.

Reply:
Good point, I'll add that as soon as I get a chance